### PR TITLE
Refactored MongoDB connection to only initiate when required

### PR DESCRIPTION
**Pull request by Sarthak027**
### Task completed

- [X] Modify the existing code to connect to the MongoDB database only when a request is received that requires database access.

- [X] Ensure that the database connection is closed when it is no longer needed to conserve resources.

- [X] Implement a mechanism to handle errors that may occur when connecting to the database lazily.

This PR solves issue #18

### CODE REVIEW

File: server/controller/urlController.js [hunk 1]
```diff
@@ -1,67 +1,79 @@
...
 
-dotenv.config();
+dotenv.config(); 
 
+// Define asynchronous function to handle specific URL retrieval
 const getSpecificUrl = async (req, res) => {
   try {
-    //receives json data from client.
+    // Retrieve shortId from request parameters
     const shortId = req.params.shortId;
```
Comment (abhijeetnishal) [inline server/controller/urlController.js:13]: **Use JS destructuring to get key**
Like: 
`const { shortId } = req.params;`

File: server/controller/urlController.js [hunk 2]
```diff
@@ -1,67 +1,79 @@
...
     console.log(error);
     res.status(500).json({ message: "Internal Server Error" });
   }
 };
 
+// Define asynchronous function to create a short URL
 const createUrl = async (req, res) => {
-  //receives json data from client
+  // Retrieve original URL from request body
   const originalUrl = req.body.originalUrl;
```
Comment (abhijeetnishal) [inline server/controller/urlController.js:41]: **Here also, use the JS Destructuring**

**Comment by abhijeetnishal** [comment]
Also remove the dbConnect() method from server.js file, now don't need it.

**Comment by abhijeetnishal** [comment]
Hey @Sarthak027, I have added the required changes that needs to be done.
After these changes we are good to Merge this PR.

**Comment by Sarthak027** [comment]
- [X] Used JS destructuring to get key.
- [X] Removed the dbConnect() method from server.js file.

**Comment by Sarthak027** [comment]
Hey @abhijeetnishal  can you please add the lable Gssoc'24 in this issuer!!

**Comment by abhijeetnishal** [comment]
> Hey @abhijeetnishal can you please add the lable Gssoc'24 in this issuer!!

Hey @Sarthak027 l, I have linked the issue where I have added labels.

**Pull request merged**